In [3]:
import pandas as pd

# Replace 'file_path.txt' with your file path
data = pd.read_csv('data/inci_name.txt', delimiter = '\t') # if the file is tab-delimited

FileNotFoundError: [Errno 2] No such file or directory: 'data/inci_name.txt'

In [2]:
data

PRODUCT  \
0                                             NaN   
1                       Apple Seed Oil (SAP only)   
2                       Absolute: Bourbon Vanilla   
3                  Absolute: Bourbon Vanilla, 20X   
4            Absolute: Jasmine Auriculatum, India   
...                                           ...   
2811                             Wax: Sumac Fruit   
2812                               Wax: Sunflower   
2813                              Witch Hazel 86%   
2814  Ylang Ylang Essential Oil (Grade: Complete)   
2815                                Yogurt Powder   

                                                   INCI           PRODUCT_ID  
0                     Pistacia Vera (Pistachio) Nut Oil          oilpistit64  
1                            Pyrus Malus Apple Seed Oil       apple seed oil  
2                       Vanilla planifolia Bean Extract        absvanillamad  
3                            Vanilla planifolia Extract         absvanilla25  
4                 Jasmine Auriculatum (Jasmine) Extract            absjasaur  
...                                                 ...                  ...  
2811                  Rhus Succedanea (Sumac) Fruit Wax  waxsumacfrdeccn1003  
2812             Helianthus Annuus (Sunflower) Seed Wax        waxsunflcn761  
2813  Hamamelis Virginiana (Witch Hazel) Water (and)...       othwithaz86us9  
2814                  Cananga Odorata (Ylang Ylang) Oil          eoyycmg1100  
2815                                      Yogurt Powder        yogurtpwdus53  

[2816 rows x 3 columns]

In [3]:
# Extract the 'INCI' column
column = data['INCI']

# Convert the column to a list
column_list = column.tolist()

# Convert the list to a Series
column_series = pd.Series(column_list)

column_series

0                       Pistacia Vera (Pistachio) Nut Oil
1                              Pyrus Malus Apple Seed Oil
2                         Vanilla planifolia Bean Extract
3                              Vanilla planifolia Extract
4                   Jasmine Auriculatum (Jasmine) Extract
                              ...                        
2811                    Rhus Succedanea (Sumac) Fruit Wax
2812               Helianthus Annuus (Sunflower) Seed Wax
2813    Hamamelis Virginiana (Witch Hazel) Water (and)...
2814                    Cananga Odorata (Ylang Ylang) Oil
2815                                        Yogurt Powder
Length: 2816, dtype: object

**Note:** Unfortunately i've only noticied that we got lot's of "Fragrance" repeating itself only after exporting our JSON and running the script for the first time, need to see why that's happening!

In [4]:
# Count the occurrences of each value
value_counts = column_series.value_counts()

# Print the count for "Fragrance"
print(value_counts["Fragrance"])

922


In [5]:
filtered_df = data[data['INCI'] == 'Fragrance']

filtered_df

PRODUCT       INCI  \
14                         Arabian Fragrance Oil: Abal  Fragrance   
15                         Arabian Fragrance Oil: Isra  Fragrance   
16                       Arabian Fragrance Oil: Jannah  Fragrance   
17                        Arabian Fragrance Oil: Motia  Fragrance   
18                 Arabian Fragrance Oil: Musk Ghazala  Fragrance   
...                                                ...        ...   
2777                           Test Product: Number 24  Fragrance   
2779                            Test Product: Number 8  Fragrance   
2791  Water-Soluble Fragrance: Cotton Blossom BBW Type  Fragrance   
2792                Water-Soluble Fragrance: Green Tea  Fragrance   
2793           Water-Soluble Fragrance: Orange Blossom  Fragrance   

          PRODUCT_ID  
14           afoabal  
15           afoisra  
16         afojannah  
17          afomotia  
18    afomuskghazala  
...              ...  
2777   testproduct24  
2779    testproduct8  
2791      fwcotbloss  
2792  fwgreenteaus14  
2793    fworangeus14  

[922 rows x 3 columns]

That's why. So what i'll need to do later on is to save all the values of product afther the ":" in another dataframe, then drop all the rows with value "Fragrance" in INCI, then append with the initial result to get a better value of names and then run again the program with this one.

In [6]:
# Split the 'Product' column at ':', expand=True splits the string into different columns
split_df = filtered_df['PRODUCT'].str.split(':', expand=True)

split_df

0                         1
14      Arabian Fragrance Oil                      Abal
15      Arabian Fragrance Oil                      Isra
16      Arabian Fragrance Oil                    Jannah
17      Arabian Fragrance Oil                     Motia
18      Arabian Fragrance Oil              Musk Ghazala
...                       ...                       ...
2777             Test Product                 Number 24
2779             Test Product                  Number 8
2791  Water-Soluble Fragrance   Cotton Blossom BBW Type
2792  Water-Soluble Fragrance                 Green Tea
2793  Water-Soluble Fragrance            Orange Blossom

[922 rows x 2 columns]

In [7]:
# Select the second column (index 1) which is the part after ':'
filtered_df['Product'] = split_df[1].str.strip()  # str.strip() is used to remove leading and trailing whitespace

filtered_df

C:\Users\deimo\AppData\Local\Temp\ipykernel_26216\1775019499.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df['Product'] = split_df[1].str.strip()  # str.strip() is used to remove leading and trailing whitespace


PRODUCT       INCI  \
14                         Arabian Fragrance Oil: Abal  Fragrance   
15                         Arabian Fragrance Oil: Isra  Fragrance   
16                       Arabian Fragrance Oil: Jannah  Fragrance   
17                        Arabian Fragrance Oil: Motia  Fragrance   
18                 Arabian Fragrance Oil: Musk Ghazala  Fragrance   
...                                                ...        ...   
2777                           Test Product: Number 24  Fragrance   
2779                            Test Product: Number 8  Fragrance   
2791  Water-Soluble Fragrance: Cotton Blossom BBW Type  Fragrance   
2792                Water-Soluble Fragrance: Green Tea  Fragrance   
2793           Water-Soluble Fragrance: Orange Blossom  Fragrance   

          PRODUCT_ID                  Product  
14           afoabal                     Abal  
15           afoisra                     Isra  
16         afojannah                   Jannah  
17          afomotia                    Motia  
18    afomuskghazala             Musk Ghazala  
...              ...                      ...  
2777   testproduct24                Number 24  
2779    testproduct8                 Number 8  
2791      fwcotbloss  Cotton Blossom BBW Type  
2792  fwgreenteaus14                Green Tea  
2793    fworangeus14           Orange Blossom  

[922 rows x 4 columns]

**Note:** Now that we have only the Product name, let's get those and put into a list like the first part.

In [8]:
# Extract the 'INCI' column
column2 = filtered_df['Product']

# Convert the column to a list
column_list2 = column2.tolist()

# Convert the list to a Series
column_series2 = pd.Series(column_list2)

column_series2

0                         Abal
1                         Isra
2                       Jannah
3                        Motia
4                 Musk Ghazala
                ...           
917                  Number 24
918                   Number 8
919    Cotton Blossom BBW Type
920                  Green Tea
921             Orange Blossom
Length: 922, dtype: object

**Note:** Now i need to remove all the Fragance values from the first column_series and then add the column_series2

In [9]:
# Drop all the values "Fragrance" from column_series
column_series = column_series[column_series != 'Fragrance']

column_series

0                       Pistacia Vera (Pistachio) Nut Oil
1                              Pyrus Malus Apple Seed Oil
2                         Vanilla planifolia Bean Extract
3                              Vanilla planifolia Extract
4                   Jasmine Auriculatum (Jasmine) Extract
                              ...                        
2811                    Rhus Succedanea (Sumac) Fruit Wax
2812               Helianthus Annuus (Sunflower) Seed Wax
2813    Hamamelis Virginiana (Witch Hazel) Water (and)...
2814                    Cananga Odorata (Ylang Ylang) Oil
2815                                        Yogurt Powder
Length: 1894, dtype: object

In [10]:
# Append column_series2 to column_series, not append, it's concact()
# Concatenate column_series and column_series2
column_series = pd.concat([column_series, column_series2], ignore_index=True)

column_series

0           Pistacia Vera (Pistachio) Nut Oil
1                  Pyrus Malus Apple Seed Oil
2             Vanilla planifolia Bean Extract
3                  Vanilla planifolia Extract
4       Jasmine Auriculatum (Jasmine) Extract
                        ...                  
2811                                Number 24
2812                                 Number 8
2813                  Cotton Blossom BBW Type
2814                                Green Tea
2815                           Orange Blossom
Length: 2816, dtype: object

**Note:** Now it's to export this series and then run the initial.py again

In [11]:
# Export the Series to a JSON file
# Replace 'file_path.json' with your file path
column_series.to_json('data/INCI_names.json', orient='records')

**Note:** initial.py finished running with 2816 requests and now i want to analyze the results in here to see what i can expect after the extraction

In [17]:
import json

# Load the JSON file
with open('data/INCI_results.json', 'r') as f:
    data = json.load(f)

# Convert the data into a pandas DataFrame
df = pd.DataFrame(data)

# Create a new DataFrame with just the 'name' and 'url' columns
df_name_url = df[['name', 'url']]

df_name_url

name  \
0                            PISTACIA LENTISCUS FRUIT OIL   
1                                  PISTACIA LENTISCUS GUM   
2                              PISTACIA LENTISCUS GUM OIL   
3                            PISTACIA LENTISCUS GUM WATER   
4                            PISTACIA LENTISCUS LEAF CERA   
...                                                   ...   
227231  SACCHAROMYCES/BITTER ORANGE FRUIT FERMENT EXTR...   
227232  SACCHAROMYCES/LACTOBACILLUS/MANDARIN ORANGE PE...   
227233  SCHIZOPHYLLUM COMMUNE/CITRUS UNSHIU PEEL EXTRA...   
227234                    SOLANUM QUITOENSE FRUIT EXTRACT   
227235                      SOLANUM QUITOENSE FRUIT JUICE   

                                                      url  
0       https://cosmileeurope.eu/inci/detail/23475/pis...  
1       https://cosmileeurope.eu/inci/detail/11823/pis...  
2       https://cosmileeurope.eu/inci/detail/17213/pis...  
3       https://cosmileeurope.eu/inci/detail/17214/pis...  
4       https://cosmileeurope.eu/inci/detail/11826/pis...  
...                                                   ...  
227231  https://cosmileeurope.eu/inci/detail/29786/sac...  
227232  https://cosmileeurope.eu/inci/detail/14001/sac...  
227233  https://cosmileeurope.eu/inci/detail/19630/sch...  
227234  https://cosmileeurope.eu/inci/detail/15223/sol...  
227235  https://cosmileeurope.eu/inci/detail/15224/sol...  

[227236 rows x 2 columns]

**Note:** 2816 requests provided 227236 urls, something is strange, probably there's a lots of duplicated data, let's take a look!

In [22]:
# Drop duplicates
df_name_url_unique = df_name_url.drop_duplicates()

# Display the DataFrame
display(df_name_url_unique)

name  \
0               PISTACIA LENTISCUS FRUIT OIL   
1                     PISTACIA LENTISCUS GUM   
2                 PISTACIA LENTISCUS GUM OIL   
3               PISTACIA LENTISCUS GUM WATER   
4               PISTACIA LENTISCUS LEAF CERA   
...                                      ...   
213652            DIOSPYROS KAKI LEAF POWDER   
213654           DIOSPYROS KAKI PEEL EXTRACT   
225641      AFRAMOMUM ANGUSTIFOLIUM LEAF OIL   
225642  AFRAMOMUM ANGUSTIFOLIUM SEED EXTRACT   
225644         AMOMUM SUBULATUM SEED EXTRACT   

                                                      url  
0       https://cosmileeurope.eu/inci/detail/23475/pis...  
1       https://cosmileeurope.eu/inci/detail/11823/pis...  
2       https://cosmileeurope.eu/inci/detail/17213/pis...  
3       https://cosmileeurope.eu/inci/detail/17214/pis...  
4       https://cosmileeurope.eu/inci/detail/11826/pis...  
...                                                   ...  
213652  https://cosmileeurope.eu/inci/detail/18975/dio...  
213654  https://cosmileeurope.eu/inci/detail/4799/dios...  
225641  https://cosmileeurope.eu/inci/detail/530/afram...  
225642  https://cosmileeurope.eu/inci/detail/531/afram...  
225644  https://cosmileeurope.eu/inci/detail/958/amomu...  

[14848 rows x 2 columns]

2815 requests provided 14848 valid urls without any duplicated, the databased supposedly has 30.000 data, for a initial Scrap, that's a good result!

**Note:** Cleaned `INCI_info.csv` manually and saved as `INCI_info_adjusted`, gonna create later a function to clean it at the same time as the extract process happens.